torchao/quantization/qat/linear.py

```diff
@@ -102,7 +102,7 @@ def forward(self, x: torch.Tensor) -> torch.Tensor:
             w = self.weight_fake_quantizer(self.weight)
         else:
             w = self.weight
-        return F.linear(x, w, self.bias)
+        return F.linear(x.to(w.dtype), w, self.bias)
```

Comment:
is this change needed?

Reply:
i am completely flummoxed by this error i'm seeing without it. Somehow these changes broke sparsity, qat and low bit optimizers without actually touching any of their code. I'm at a loss as to how and trying to figure out whats going on. If i can figure out how to fix it without that then i'll do that but thus far that actually seemed to have fixed it.